catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/content/ContentProducerDataAccessObject.java

```diff
@@ -148,13 +159,41 @@ public void createContentItem(FileSystemPersistenceProvider fileIdMap, ContentEn
                 }
             }
         } else if (StandardWatchEventKinds.ENTRY_MODIFY.equals(eventType)) {
+
+            Metacard metacard = null;
+
+            if (StringUtils.isNotEmpty(id)) {
```

Comment:
If any of these checks fail, we will fall back to the previous behavior. Shouldn't we throw an exception instead?